src/pages/cw-storage-plus/containers/map.mdx

```diff
@@ -161,3 +162,139 @@ we lock the first component to `"alice"`, entries are ordered lexicographically
 component.
 
 </Callout>
+
+## Advanced
+
+### Using custom types as a Key in a Map storage
+
+The current section provides an example of an implementation for
+[`PrimaryKey`](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/trait.PrimaryKey.html)
+(and `KeyDeserialize`) traits.
+
+Let's imagine that we would like to use `Denom` as a key in a map like `Map<Denom, Uint64>`.
+Denom is a widely used enum that represents either a CW20 token or a Native token.
+
+Here is a complete implementation.
+
+```rust template="storage"
```

Comment:
I can't get this example to compile. We have automation that picks up code snippets and runs them as tests.

You can give it a try locally by running `scripts/test-rust.sh`

Feel free to tinker until it compiles, or let me know if you don't have time. I'm happy to pick this up sometime soon.